### PR TITLE
cache plugins repo

### PR DESCRIPTION
**Pull request by artem-fedorov**

### CODE REVIEW

**Comment by artem-fedorov** [comment]
i'll add tests soon

File: src/main/java/org/jmeterplugins/repository/JARSourceHTTP.java [hunk 1]
```diff
@@ -109,6 +133,11 @@ private AbstractHttpClient getHTTPClient() {
     }
 
     protected JSON getJSON(String uri) throws IOException {
+        PluginsRepo repo = getRepoCache(uri);
+        if (repo != null && repo.isActual()) {
+            log.info("Found cached repo");
+            return JSONSerializer.toJSON(repo.getRepoJSON(), new JsonConfig());
+        }
 
         log.debug("Requesting " + uri);
```
Comment (artem-fedorov) [inline src/main/java/org/jmeterplugins/repository/JARSourceHTTP.java]: Can we make this log and log above INFO level?  It'll help for check that this cache is work

Comment (undera) [reply src/main/java/org/jmeterplugins/repository/JARSourceHTTP.java]: Yes, we can

File: src/main/java/org/jmeterplugins/repository/JARSourceHTTP.java [hunk 2]
```diff
@@ -147,6 +178,79 @@ protected JSON getJSON(String uri) throws IOException {
...
+        long expirationTime = date + maxAge;
+        PluginsRepo repo = new PluginsRepo(repoJSON, expirationTime);
+        SerializeUtils.serialize(repo, generateCacheFile(uri));
+    }
+
+    private File generateCacheFile(String uri) {
+        return new File(cacheDir, generateFileName(uri));
+    }
+
+    private String generateFileName(String uri) {
```
Comment (undera) [inline src/main/java/org/jmeterplugins/repository/JARSourceHTTP.java]: Why not just generate md5 from URI and don't mess with words?

Comment (undera) [reply src/main/java/org/jmeterplugins/repository/JARSourceHTTP.java]: And also add `System.getProperty("user.name")` to that MD5, so files of different users won't clash

**Comment by artem-fedorov** [comment]
Should I add test for HTTP headers?

**Comment by undera** [comment]
It's all covered, as I see

**Pull request merged**